### ISSUE TITLE
Filtering on edge property returns NO results

### ISSUE THREAD

**Issue by grachit** [orientechnologies/orientdb#5451]
using orient db 2.1.6

Running following query returns no result:
```sql
select expand(out("RUNS_ON").in("EXECUTED_ON")[JobSubmitted between '2015-12-09 00:00:00' and '2015-12-09 23:59:59']) from JOBS WHERE JobID = '3'
```

**Comment by luigidellaquila** [comment]
Hi @grachit 

it seems to be a parse related issue. Just let me tell you that square bracket notation has many limitations, eg. you can not use multiple conditions in it.

I'll check it and I'll let you know

Thanks

Luigi

**Comment by grachit** [comment]
Thnaks @luigidellaquila, It does not seem to be working on simple queries too...

**Comment by grachit** [comment]
Hi,
did u get a chance to check that out?

**Comment by luigidellaquila** [comment]
Hi @grachit 

not yet, sorry, but I just noticed a thing: your goal is to filter on EDGE property, while your query is filtering on VERTEX property!

Could you please try this and let me know if it works?

```
select expand(out("RUNS_ON").inE("EXECUTED_ON")[JobSubmitted between '2015-12-09 00:00:00' and '2015-12-09 23:59:59'].outV()) from JOBS WHERE JobID = '3'
```

Thanks

Luigi

**Comment by grachit** [comment]
does not work

**Comment by grachit** [comment]
RUNS_ON is a edge running out from JOBS to another vertex (Lets say A). A has in edge called EXECUTED_ON. and this in edge has property called JobSubmitted.

what's the difference between in() and inE()?

It looks like between  or > do not work.

only = is working:
select expand(out("RUNS_ON").inE("EXECUTED_ON")[JobSubmitted = '2015-12-09 01:54:47.667'  ].outV()) from JOBS WHERE JobID = '3'

**Comment by luigidellaquila** [comment]
Yes, it makes sense. It's definitely a problem in the old executor layer.

The difference between in() and inE() is that the first one returns vertices, while the second one returns edges. 
```inE().out()``` is the same as ```in()```

Thanks

Luigi

**Comment by luigidellaquila** [comment]
sorry, I meant ```inE().outV()``` is the same as ```in()```

**Comment by grachit** [comment]
any update over it?

**Comment by luigidellaquila** [comment]
Fixed in 2.1.x and develop branches. The fix will be released with 2.1.12

Closing, please reopen if you see any unexpected behaviors

Thanks

Luigi
